src/components/editor-page/editor-pane/table-extractor.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { createNumberRangeArray } from '../../common/number-range/number-range'
+
+export const isTable = (text: string): boolean => {
+  // Tables must consist of multiple rows and columns
+  if (!text.includes('\n') || !text.includes('\t')) {
+    return false
+  }
+  // Code within code blocks should not be parsed as a table
+  if (text.startsWith('```')) {
+    return false
+  }
+
+  const lines = text.split(/\r?\n/)
+                    .filter(line => line.trim() !== '')
+
+  // Tab-indented text should not be matched as a table
+  if (lines.every(line => line.startsWith('\t'))) {
+    return false
+  }
+  // Every line should have the same amount of tabs (table columns)
+  const tabsPerLines = lines.map(line => line.match(/\t/g)?.length ?? 0)
+  return tabsPerLines.every(line => line === tabsPerLines[0])
+}
+
+export const clipboardTableToMarkdown = (pasteData: string): string => {
```

Comment:
Clipboard is a weird verb :yum:

Reply:
May I suggest `convertCclipboardTableToMarkdown`

Reply:
nope. to many carbon copies.

Reply:
`cC`

Reply:
Changed